### PR TITLE
tests: share http server between tests

### PR DESCRIPTION
**Pull request by bartlomieju**
This is attempt to fix flaky tests related to `019_media_types.ts`. 

I believe the problem sprung from http server used in tests that was created and killed between each test case. This PR allows to share http server between tests.

Ref #3332
CC @ry

### CODE REVIEW

**Comment by ry** [comment]
@bartlomieju Do you have a theory why restarting the http server would cause failures like this?

File: cli/test_util.rs [hunk 1]
```diff
@@ -35,45 +37,52 @@ pub fn deno_exe_path() -> PathBuf {
...
+    if count == 1 {
+      kill_http_server();
     }
   }
 }
 
-/// Starts tools/http_server.py when the returned guard is dropped, the server
-/// will be killed.
-pub fn http_server<'a>() -> HttpServerGuard<'a> {
-  // TODO(ry) Allow tests to use the http server in parallel.
```
Comment (ry) [inline cli/test_util.rs]: 👍

**Comment by bartlomieju** [comment]
> @bartlomieju Do you have a theory why restarting the http server would cause failures like this?

Not really, those failures resemble very closely situation when you try running `lock_err` test without spawned http server (connection refused error). I'm not even sure if that's possible but that suggest that server was not running from any reason. Let's try with this solution and see if problems reappear?

**Comment by ry** [comment]
There seems to be a failure here https://github.com/denoland/deno/runs/302044357

**Comment by bartlomieju** [comment]
I'll investigate further tomorrow, also I'll try to add some debug machinery to CI so we don't look in the dark

**Pull request merged**